From 21dacdd4d67458fe54ac05c15b58bb7a96a69f25 Mon Sep 17 00:00:00 2001
From: Tracy Sharpe <42477615+tracysh@users.noreply.github.com>
Date: Fri, 22 Mar 2019 18:13:41 -0700
Subject: [PATCH] fix epilogue code (#695)

Repair the epilogue code for the aarch64 SGEMM kernel to restore registers from the correct stack location.
---
 onnxruntime/core/mlas/lib/aarch64/sgemma.s | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/onnxruntime/core/mlas/lib/aarch64/sgemma.s b/onnxruntime/core/mlas/lib/aarch64/sgemma.s
index d6409295..545465a5 100644
--- a/onnxruntime/core/mlas/lib/aarch64/sgemma.s
+++ b/onnxruntime/core/mlas/lib/aarch64/sgemma.s
@@ -453,7 +453,7 @@ MlasSgemmKernel\Mode\():
 .L\Mode\().ExitKernel:
         mov     x0,x4
         ldp     d10,d11,[sp,16]
-        ldp     d8,d9,[sp,32]!
+        ldp     d8,d9,[sp],32
         ret
 
 //
-- 
2.16.1.windows.4

